### PR TITLE
Numerous documentation tweaks and improvements

### PR DESCRIPTION
**Pull request by mattcaswell**
Updates to various README files - in part inspired by @jamuir's experience described in https://github.com/openssl/project/issues/253

### CODE REVIEW

**Comment by t8m** [comment]
CI is relevant

File: README-QUIC.md [hunk 1]
```diff
@@ -83,16 +57,26 @@ an existing application.
...
+HTTP/3 so connecting to an HTTP/3 server should be possible but sending an
+HTTP/3 request or receiving any response data is not.
 
 [openssl-quic(7) manual page]: https://www.openssl.org/docs/manmaster/man7/openssl-quic.html
 [OpenSSL guide]: https://www.openssl.org/docs/manmaster/man7/ossl-guide-introduction.html
 [DDD]: https://github.com/openssl/openssl/tree/master/doc/designs/ddd
 [found in the source tree under `doc/designs/ddd`]: ./doc/designs/ddd/
 [demo found in `demos/http3`]: ./demos/http3/
-[openssl-quic-background(7) manual page]: https://www.openssl.org/docs/manmaster/man7/openssl-quic-background.html
+[openssl-quic(7) manual page]: https://www.openssl.org/docs/manmaster/man7/openssl-quic.html
```
Comment (t8m) [inline README-QUIC.md:79]: A duplicate of line 74

File: README-QUIC.md [hunk 2]
```diff
@@ -7,12 +7,14 @@ client.
 Users interested in using the new QUIC functionality are encouraged to look at
 some of the following resources:
 
-- The [openssl-quic(7) manual page], which provides a basic reference overview
-  of QUIC functionality and how use of QUIC differs from use of TLS with regard
-  to our API;
 - The new [OpenSSL Guide], which provides introductory guides
   on the use of TLS, QUIC, and other OpenSSL functionality. See the
-  [ossl-guide-introduction(7) manual page][OpenSSL Guide] for the index.
+  [ossl-guide-introduction(7) manual page] for the index.
```
Comment (t8m) [inline README-QUIC.md:12]: `[OpenSSL Guide]` is an existing link but `[ossl-guide-introduction(7) manual page]` does not exist? But they would be supposedly the same links?

File: README-QUIC.md [hunk 3]
```diff
@@ -7,12 +7,14 @@ client.
...
+- The [OpenSSL Guide] incorporates various code samples. The complete source
+  for these can be [found in the source tree under `demos\guide`](./demos/guide/).
```
Comment (t8m) [inline README-QUIC.md:14]: `demos/guide` would be better wouldn't it?

File: README-QUIC.md [hunk 4]
```diff
@@ -83,16 +57,26 @@ an existing application.
...
+$ openssl s_client -quic -alpn myalpn -connect host:port
 ```
 
-This connects to a QUIC server using the specified ALPN protocol name and opens
-a single bidirectional stream. Data can be passed via stdin/stdout as usual.
-This allows test usage of QUIC using simple TCP/TLS-like usage.
+In the above example replace `host` with the hostname of the server (e.g.
+`www.example.com`) and `port` with the port for the server (e.g. `443`). Replace
+`myalpn` with the Application Layer Protocol to use (e.g.`h3` represents
+HTTP/3). IANA matains a standard list of [ALPN ids] that can be used.
```
Comment (t8m) [inline README-QUIC.md:66]: Typo: maintains

File: README.md [hunk 5]
```diff
@@ -7,10 +7,11 @@ Welcome to the OpenSSL Project
 [![appveyor badge]][appveyor jobs]
 
 OpenSSL is a robust, commercial-grade, full-featured Open Source Toolkit
-for the Transport Layer Security (TLS) protocol formerly known as the
-Secure Sockets Layer (SSL) protocol. The protocol implementation is based
-on a full-strength general purpose cryptographic library, which can also
-be used stand-alone.
+for the TLS (formerly SSL), DTLS and QUIC (client side only) protocols.
```
Comment (t8m) [inline README.md:10]: maybe:  `currently client side only`

**Review by t8m — CHANGES_REQUESTED** [review]

**Comment by mattcaswell** [comment]
All feedback addressed. Please take another look.

File: README-QUIC.md [hunk 6]
```diff
@@ -25,43 +26,15 @@ FAQ
...
-For more background information on OpenSSL's QUIC implementation, see the
+serve many of the use cases of SSL/TLS as well as those of DTLS.
+
+QUIC delivers a number of advantages such as support for multiple streams of
+communication; it is the basis for HTTP/3 [RFC 9114]; fast connection
+initiation; and connection migration (enabling a connection to survive IP
+address changes). For a more complete description of what QUIC is and its
+advantages see the [QUIC Introduction] in the [OpenSSL Guide].
+
+For a more comprehensive overview of OpenSSL's QUIC implementation, see the
```
Comment (jamuir) [inline README-QUIC.md:37]: I think this sentence reads better if it does not start in the same way as the previous sentence.

maybe reword and drop "more":  "For a comprehensive overview...."

File: README.md [hunk 7]
```diff
@@ -35,7 +37,9 @@ Overview
 The OpenSSL toolkit includes:
 
 - **libssl**
-  an implementation of all TLS protocol versions up to TLSv1.3 ([RFC 8446]).
+  an implementation of all TLS protocol versions up to TLSv1.3 ([RFC 8446]),
+  DTLS protocol versions up to DTLSv1.2 ([RFC 6347]) and
+  the QUIC (client side only) version 1 protocol ([RFC 9000]).
```
Comment (jamuir) [inline README.md:42]: should we say "currently client side only" as before?

**Comment by mattcaswell** [comment]
Latest round of feedback addressed. Please take another look.

**Review by hlandau — APPROVED** [review]

**Review by t8m — APPROVED** [review]

**Comment by hlandau** [comment]
Merged to master and 3.2. Thank you.
